File: example/Monica_SoilTemp/src/cs/Monica_SoilTemp/SoilTemperatureCompExogenous.cs
using System;
using System.Collections.Generic;

public class SoilTemperatureCompExogenous 
{
    private double _tmin;
    private double _tmax;
    private double _globrad;
    private double _soilCoverage;
    private double _soilSurfaceTemperatureBelowSnow;
    private bool _hasSnowCover;
    
        public SoilTemperatureCompExogenous() { }
    
    
    public SoilTemperatureCompExogenous(SoilTemperatureCompExogenous toCopy, bool copyAll) // copy constructor 
    {
    if (copyAll)
    {
    
    tmin = toCopy.tmin;
    tmax = toCopy.tmax;
    globrad = toCopy.globrad;
    soilCoverage = toCopy.soilCoverage;
    soilSurfaceTemperatureBelowSnow = toCopy.soilSurfaceTemperatureBelowSnow;
    hasSnowCover = toCopy.hasSnowCover;
    }
    }
    public double tmin
        {
            get { return this._tmin; }
            set { this._tmin= value; } 
        }
    public double tmax
        {
            get { return this._tmax; }
            set { this._tmax= value; } 
        }
    public double globrad
        {
            get { return this._globrad; }
            set { this._globrad= value; } 
        }
    public double soilCoverage
        {
            get { return this._soilCoverage; }
            set { this._soilCoverage= value; } 
        }
    public double soilSurfaceTemperatureBelowSnow
        {
            get { return this._soilSurfaceTemperatureBelowSnow; }
            set { this._soilSurfaceTemperatureBelowSnow= value; } 
        }
    public bool hasSnowCover
        {
            get { return this._hasSnowCover; }
            set { this._hasSnowCover= value; } 
        }
}

File: example/Monica_SoilTemp/src/java/Monica_SoilTemp/SoilTemperatureCompExogenous.java
import  java.io.*;
import  java.util.*;
import java.time.LocalDateTime;

public class SoilTemperatureCompExogenous
{
    private double tmin;
    private double tmax;
    private double globrad;
    private double soilCoverage;
    private double soilSurfaceTemperatureBelowSnow;
    private Boolean hasSnowCover;
    
    public SoilTemperatureCompExogenous() { }
    
    public SoilTemperatureCompExogenous(SoilTemperatureCompExogenous toCopy, boolean copyAll) // copy constructor 
    {
        if (copyAll)
        {
            this.tmin = toCopy.gettmin();
            this.tmax = toCopy.gettmax();
            this.globrad = toCopy.getglobrad();
            this.soilCoverage = toCopy.getsoilCoverage();
            this.soilSurfaceTemperatureBelowSnow = toCopy.getsoilSurfaceTemperatureBelowSnow();
            this.hasSnowCover = toCopy.gethasSnowCover();
        }
    }
    public double gettmin()
    { return tmin; }

    public void settmin(double _tmin)
    { this.tmin= _tmin; } 
    
    public double gettmax()
    { return tmax; }

    public void settmax(double _tmax)
    { this.tmax= _tmax; } 
    
    public double getglobrad()
    { return globrad; }

    public void setglobrad(double _globrad)
    { this.globrad= _globrad; } 
    
    public double getsoilCoverage()
    { return soilCoverage; }

    public void setsoilCoverage(double _soilCoverage)
    { this.soilCoverage= _soilCoverage; } 
    
    public double getsoilSurfaceTemperatureBelowSnow()
    { return soilSurfaceTemperatureBelowSnow; }

    public void setsoilSurfaceTemperatureBelowSnow(double _soilSurfaceTemperatureBelowSnow)
    { this.soilSurfaceTemperatureBelowSnow= _soilSurfaceTemperatureBelowSnow; } 
    
    public Boolean gethasSnowCover()
    { return hasSnowCover; }

    public void sethasSnowCover(Boolean _hasSnowCover)
    { this.hasSnowCover= _hasSnowCover; } 
    
}